init.lua

```diff
@@ -1,116 +1,540 @@
-local original_pos = {}
+-- NOTE: in the output texts, the names are always in double quotes because some players have
+--	names that can be confusing without the quotes.
+-- IDEA: technically it would be possible to chain observe. Would have to climb the parent tree
+--	making sure there is nothing circular happening. Including checking all the children.
+--	A lot can go wrong with that, so it has been left out for now.
+--	Another complication to this is that there are many combinations of client<->server
+--	software versions to consider.
+-- IDEA: might be nice to have a /send_home (<player>|all) command for invitie to detach
+--	invited guests again.
+--	Currently player can force detachment by logging off.
+spectator_mode = {
+	version = 20220214,
+	command_accept = minetest.settings:get('spectator_mode.command_accept') or 'smy',
+	command_deny = minetest.settings:get('spectator_mode.command_deny') or 'smn',
+	command_detach = minetest.settings:get('spectator_mode.command_detach') or 'unwatch',
+	command_invite = minetest.settings:get('spectator_mode.command_invite') or 'watchme',
+	command_attach = minetest.settings:get('spectator_mode.command_attach') or 'watch',
+	invitation_timeout = tonumber(minetest.settings:get(
+		'spectator_mode.invitation_timeout') or 1 * 60),
+
+	keep_all_observers_alive = minetest.settings:get_bool(
+		'spectator_mode.keep_all_observers_alive', false),
+
+	priv_invite = minetest.settings:get('spectator_mode.priv_invite') or 'interact',
+	priv_watch = minetest.settings:get('spectator_mode.priv_watch') or 'watch',
+}
+local sm = spectator_mode
+do
+	local temp = minetest.settings:get('spectator_mode.extra_observe_privs') or ''
+	sm.extra_observe_privs, sm.extra_observe_privs_moderator = {}, nil
+	for _, priv in ipairs(temp:split(',')) do
+		sm.extra_observe_privs[priv] = true
+	end
+	temp = minetest.settings:get('spectator_mode.extra_observe_privs_moderator')
+	if (not temp) or ('' == temp) then
+		sm.extra_observe_privs_moderator = sm.extra_observe_privs
+	else
+		sm.extra_observe_privs_moderator = {}
```

Comment:
In the code block above, `sm.extra_observe_privs` is reference-copied. Why don't you need a copy of its values here?

Reply:
Because not adding any temp-privs for moderators makes both types of players be treated equally on that matter.
If server sets moderator temp-privs then two *different* sets of privs are used.
-The first when player is invited
-the second when moderator uses '/watch'